crest/Assets/Crest/Crest/Shaders/OceanInputs/Resources/ClipSurfaceConvexHull.shader

```diff
@@ -44,8 +45,12 @@ Shader "Crest/Inputs/Clip Surface/Convex Hull"
 			Varyings Vert(Attributes input)
 			{
 				Varyings o;
-				o.positionCS = UnityObjectToClipPos(input.positionOS);
-				o.positionWS = mul(unity_ObjectToWorld, float4(input.positionOS, 1.0));
+
+				o.positionWS = mul(unity_ObjectToWorld, float4(input.positionOS, 1.0)).xyz;
+				// Correct for displacement
+				o.positionWS.xz -= _DisplacementAtInputPosition.xz;
```

Comment:
This doesn't work as expected for clip surface since it already has displacement correction in the ocean renderer.

Reply:
Oh good catch. It already samples the clip data at the displaced position. So you mean we could make it sample at the undisplaced position to make it consistent with how the other ocean data is sampled? I could try that.

Reply:
Yes. I think it makes most sense to go that route. Easier to keep track of.

Reply:
This has not really worked out. The correction this branch applies is only to the center of the input, not over the whole area. This means the clipping is inaccurate if we use this correction.

Think i will leave the ocean shader how it is, and remove this correction option for clipping inputs.

Reply:
No worries. Sounds good.